ebos/eclmpiserializer.hh

```diff
@@ -95,12 +95,16 @@ public:
             variant(data);
         } else if constexpr (is_optional<T>::value) {
             optional(data);
+        } else if constexpr (is_vector<T>::value) {
+            vector(const_cast<T&>(data));
         } else if constexpr (is_map<T>::value) {
             map(const_cast<T&>(data));
         } else if constexpr (is_array<T>::value) {
             array(const_cast<T&>(data));
         } else if constexpr (is_set<T>::value) {
             set(const_cast<T&>(data));
+        } else if constexpr (has_serializeOp<detail::remove_cvr_t<T>>::value) {
```

Comment:
Would have thought that this removes  `volatile` but it does not...

Reply:
indeed, i forgot to remove volatile.